forwarder/cmd/forwarder/forwarder.go

```diff
@@ -326,17 +290,16 @@ func fetchAndProcessLogs(ctx context.Context, storageClient *storage.Client, log
 	err = errors.Join(err, metricErr)
 
 	logger.Info(fmt.Sprintf("Finished processing %d logs", logCount))
-	return err, blobErrors
+	return err
 }
 
 func processDeadLetterQueue(ctx context.Context, now customtime.Now, logger *log.Entry, storageClient *storage.Client, logsClient *logs.Client, flushedLogsClients []*logs.Client) error {
-	// Capture context error at the start - we want to return this even if
-	// DLQ operations succeed with context recovery
-	ctxErr := ctx.Err()
+	loadAndSaveCtx, cancel := context.WithCancel(context.Background())
```

Comment:
This works, but again is a bit unexpected because `ctx` is an argument passed in but not used here. I think this should be pushed all the way up to the main function where the original context is first created.